### ISSUE TITLE
"property 'window' is a read-only and non-configurable data property " error in first test ran

### ISSUE THREAD

**Issue by jasonworden** [kaliber5/ember-window-mock#102]
Sorry, filing an issue is not the best for this, but unsure where else to post. I tried adding this in a repository that has some acceptances, integration/rendering tests, and unit tests. My first (acceptance) test fails with this error.

```
‣
TypeError: 'get' on proxy: property 'window' is a read-only and non-configurable data property on the proxy target but the proxy did not return its actual value (expected '#<Window>' but got '[object Object]')
    at isWindow (http://localhost:7357/assets/vendor.js:12508:37)
    at isArrayLike (http://localhost:7357/assets/vendor.js:12915:28)
    at Function.makeArray (http://localhost:7357/assets/vendor.js:12804:9)
    at new jQuery.fn.init (http://localhost:7357/assets/vendor.js:15430:17)
    at jQuery (http://localhost:7357/assets/vendor.js:12562:10)
    at Class.init (http://localhost:7357/assets/webapp.js:64113:7)
    at Class.superWrapper [as init] (http://localhost:7357/assets/vendor.js:65580:28)
    at Class.CoreObject (http://localhost:7357/assets/vendor.js:63012:27)
    at Class.EmberObject (http://localhost:7357/assets/vendor.js:63395:75)
    at Class (http://localhost:7357/assets/vendor.js:63074:72)
```

how can i resolve this? thx!

I'm open to making a PR to add documention to the README with the resolution of this, if applicable. 

------
ember 3.4.7
`"ember-mocha": "^0.14.0"`

-----
For setting up my acceptance tests, I tried variations of

```
import { setupWindowMock, reset as resetWindow } from 'ember-window-mock';

export const runBeforeHooks = () => {
  // see https://github.com/kaliber5/ember-window-mock#resetting-the-state-in-tests
  resetWindow();
  // setupWindowMock();
  setupRouter();
};

export const setupApplicationTest = function (testContext) {
  const hooks = mochaSetupApplicationTest();

  setupWindowMock(hooks);
  hooks.beforeEach(runBeforeHooks);

  return hooks;
};
```

**Comment by simonihmig** [comment]
Don't think this is related to test setup, but rather something in the mocked window itself goes wrong. Can you please post how your app code uses this addon? In the stack trace above it seems this is the place where you own code does something:
```
at Class.init (http://localhost:7357/assets/webapp.js:64113:7)
```

**Comment by jasonworden** [comment]
@simonihmig Thanks! Unfortunately ember tests don't provide clickable stack traces, so I had to scroll to line 64113 of my `webapp.js` to find the line. This was the problematic instance:
```
    $(window).on('beforeunload', () => {
      this.event('leavepage', { _external: true });
    });
```

Not using the mock here fixed the test. Do you know why this would be an issue?

**Comment by jasonworden** [comment]
Separately, when would I use `setupWindowMock` vs `reset` before a test?

**Comment by simonihmig** [comment]
So jQuery is accessing `window.window` [here](https://github.com/jquery/jquery/blob/e743cbd28553267f955f71ea7248377915613fd9/src/var/isWindow.js#L5), which the mock has replaced with another Proxy. This seems to be not allowed for read-only and non-configurable properties, hence the TypeError. So this can be seen as a bug here, and should be fixed! Thanks for reporting!

> Separately, when would I use setupWindowMock vs reset before a test?

I would recommend to always use `setupWindowMock`, as long as you use the new testing APIs with support for `hooks`!

**Comment by kbiedrzycki** [comment]
@simonihmig any plans to solve this? Or this is a bug in jQuery in your opinion?

**Comment by simonihmig** [comment]
Fixed in #452, confirmed in #454
